velocity_modelling/write/hdf5.py

```diff
@@ -0,0 +1,283 @@
+                f.attrs["format_version"] = "1.0"
+                f.attrs["creation_date"] = datetime.datetime.now().isoformat()
+
+                # Add all velocity model parameters from nzcvm.cfg as root attributes
+                if vm_params:
+                    config_group = f.create_group("config")
+                    for key, value in vm_params.items():
+                        # Handle special case for enum types
+                        if hasattr(value, "name"):
+                            config_group.attrs[key] = value.name
+                        elif hasattr(value, "value"):
+                            config_group.attrs[key] = value.value
+                        else:
+                            try:
+                                config_group.attrs[key] = value
+                            except TypeError:
+                                # If attribute can't be stored directly, convert to string
+                                config_group.attrs[key] = str(value)
+
+                    # Add a string representation of the original config for reference
+                    config_lines = []
+                    for key, value in vm_params.items():
+                        if hasattr(value, "name"):
+                            config_lines.append(f"{key.upper()}={value.name}")
+                        else:
+                            config_lines.append(f"{key.upper()}={value}")
+                    config_str = "\n".join(config_lines)
+                    config_group.attrs["config_string"] = config_str
+
+                # Create groups for mesh and properties
+                f.create_group("mesh")
+                f.create_group("properties")
+
+                # Create resizable datasets for properties
+                props_group = f["properties"]
+                nx, nz = partial_global_qualities.vp.shape
+
+                # Log the dimensions for debugging
+                logger.log(
+                    logging.DEBUG,
+                    f"Creating datasets with shape (nz={nz}, ny={ny}, nx={nx})",
+                )
+
+                # Create datasets with shape (nz, ny, nx): Paraview demands this order
+                props_group.create_dataset(
+                    "vp", shape=(nz, ny, nx), dtype="f4", compression="gzip"
+                )
+                props_group.create_dataset(
+                    "vs", shape=(nz, ny, nx), dtype="f4", compression="gzip"
+                )
+                props_group.create_dataset(
+                    "rho", shape=(nz, ny, nx), dtype="f4", compression="gzip"
+                )
+                props_group.create_dataset(
+                    "inbasin", shape=(nz, ny, nx), dtype="i1", compression="gzip"
+                )
+
+                # Add metadata
+                props_group["vp"].attrs["units"] = "km/s"
+                props_group["vs"].attrs["units"] = "km/s"
+                props_group["rho"].attrs["units"] = "g/cm³"
```

Comment:
Rather than using unicode I suggest `g/cm^3` because python packages that use these units attributes expect the ascii format.
```suggestion
                props_group["rho"].attrs["units"] = "g/cm^3"
```

Reply:
Actually there were a few places that had `cm³`. All updated with cm^3